### ISSUE TITLE
RuleActionOverrides throws E3002 Invalid Property with ManagedRuleGroupStatement

### ISSUE THREAD

**Issue by spedersen-emailage** [aws-cloudformation/cfn-lint#2524]
### CloudFormation Lint Version

cfn-lint 0.72.2

### What operating system are you using?

Mac

### Describe the bug

`RuleActionOverrides` throws an E3002 Invalid Property error when used with `ManagedRuleGroupStatement`.

```
$ cfn-lint ruleactionoverride.yaml 
E3002 Invalid Property Resources/ExampleWebACL/Properties/Rules/0/Statement/ManagedRuleGroupStatement/RuleActionOverrides
ruleactionoverride.yaml:30:15
```

### Expected behavior

This should not throw an error as it's valid per CloudFormation's documentation. I also successfully ran a CloudFormation stack update with a template similar to the example included in this issue.

- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-webacl.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-rule.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-statement.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-managedrulegroupstatement.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-ruleactionoverride.html

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: "RuleActionOverrides Example"

Resources:

  ExampleWebACL:
    Type: AWS::WAFv2::WebACL
    Properties:
      Name: ExampleWebACL
      DefaultAction:
        Block: {}
      Scope: REGIONAL
      VisibilityConfig:
        SampledRequestsEnabled: true
        CloudWatchMetricsEnabled: true
        MetricName: ExampleWebACL
      Rules:
        - Name: ExampleRuleActionOverrides
          Priority: 0
          OverrideAction:
            None: {}
          VisibilityConfig:
            SampledRequestsEnabled: true
            CloudWatchMetricsEnabled: true
            MetricName: ExampleRuleActionOverrides
          Statement:
            ManagedRuleGroupStatement:
              Name: AWSManagedRulesCommonRuleSet
              VendorName: AWS
              RuleActionOverrides:
                - Name: NoUserAgent_HEADER
                  ActionToUse:
                    Count: {}
```

**Comment by kddejong** [comment]
Researching:
- [CloudFormation specs](https://dnwj8swjjbsbt.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json) don't have it.

**Comment by spedersen-emailage** [comment]
> Researching:
> 
>  * [CloudFormation specs](https://dnwj8swjjbsbt.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json) don't have it.  [Main docs link](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html)

Interesting...

* AWS::WAFv2::WebACL.ManagedRuleGroupStatement
	* Properties:
		* VendorName:
		* Version:
		* ManagedRuleGroupConfigs:
		* ExcludedRules:
		* Name:
		* ScopeDownStatement:

It's documented:

* [AWS::WAFv2::WebACL ManagedRuleGroupStatement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-managedrulegroupstatement.html)

```yaml
ExcludedRules: 
  - ExcludedRule
ManagedRuleGroupConfigs: 
  - ManagedRuleGroupConfig
Name: String
RuleActionOverrides: 
  - RuleActionOverride
ScopeDownStatement: 
  Statement
VendorName: String
Version: String
```

* [AWS::WAFv2::WebACL RuleActionOverride](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-wafv2-webacl-ruleactionoverride.html)

```yaml
ActionToUse: 
  RuleAction
Name: String
```

...and works in CloudFormation. I successfully set multiple overrides for `AWSManagedRulesCommonRuleSet`.

**Comment by kddejong** [comment]
I'm going to watch this weeks spec drop to see if its covered and if not I will patch this property in so its covered going forward.

**Comment by kddejong** [comment]
This automated PR #2523 will include this property

fix #2523

**Comment by spedersen-emailage** [comment]
@kddejong Awesome! Do you know when the next release will happen? I looked at the history and it doesn't look like it's on any particular schedule. The last one was 5 hours ago, so these two fixes just missed it.

**Comment by kddejong** [comment]
Doing another one today since its just a spec update.

**Comment by kddejong** [comment]
https://github.com/aws-cloudformation/cfn-lint/releases/tag/v0.72.5

**Comment by spedersen-emailage** [comment]
Confirmed working!

```
$ cfn-lint -v
cfn-lint 0.72.5
$ cfn-lint waf_example.yaml 
$ 
```
